plans/switch_master.pp

```diff
@@ -0,0 +1,47 @@
+plan peadm::switch_master (
+  TargetSpec $nodes,
+  String $master,
+  Boolean $sign_csr = false,
+
+){
+
+  # Extract the target node name and convert to certname array
+   $certname = get_targets($nodes).map |$a| { $a.name }
+
+  # Task to stop Puppet and change the Puppet Server name in puppet.conf
+    run_task ('peadm::agent_switchover',$nodes,
+      master => $master
+  )
+
+  # If the client does not have csr_attributes.yaml run the plan with sign_csr=true
+    if $sign_csr == true {
+
+    # Request a certificate from the master this command will fail so we set _catch_errors,
+    # this will allow the plan to run without exiting with an error
+      run_task('peadm::puppet_runonce', $nodes,
```

Comment:
Use `peadm::submit_csr`. If you need it to work with Windows, update `peadm::submit_csr` to support it. The only change necessary should be making the fully-qualified path to Puppet a variable and setting it appropriately based on the OS.

Reply:
@reidmv is there a reason peadm does not use puppetlabs/pe_boostrap for the agent install? as it supports CSRs etc? 
i asked @chlawren to take a look at the module, but if you have 'reasons' you use an independent task for agent install and csr that's valid too.

Reply:
For PE provisioning, the peadm module carefully separates the agent installation from running the agent, and from submitting a certificate signing request. Because the agent doesn't run when installed, there is no CSR. Because when we want a CSR we don't yet want the agent to run, we avoid running it just to get the CSR as a side-effect; we _only_ submit a CSR. We run the agent later.

I believe I originally chose not to use bootstrap because I determined that 72 of the 76 lines of code in the task were unnecessary, and I didn't even like the parameter interface. The agent install task peadm currently uses is four lines of code (three lines of whitespace), and the parameter interface is a straight pass-through to interacting directly with the agent install script provided by Puppet enterprise. Fewer layers of abstraction and one less dependency on a module that doesn't actually accomplish much.